### PR TITLE
Add note about special handling of 'name' argument

### PR DESCRIPTION
**Pull request by melkamar**
Fixes https://github.com/symfony/symfony-docs/issues/13422

### CODE REVIEW

File: components/config/definition.rst [hunk 1]
```diff
@@ -346,6 +346,10 @@ In order to maintain the array keys use the ``useAttributeAsKey()`` method::
         ->end()
     ;
 
+.. note::
+
+    If using YAML, then the ``'name'`` argument of ``useAttributeAsKey`` has a special meaning and refers to the key of the map (``sf_connection`` and ``default`` in this example). If a child node was defined for the ``connections`` node with the name ``name``, then the key of the map would be lost.
```
Comment (dbrumann) [inline components/config/definition.rst]:
```suggestion
    If using YAML, then the ``'name'`` argument of ``useAttributeAsKey`` has a special meaning and refers to the key of the map (``sf_connection`` and ``default`` in this example). If a child node was defined for the ``connections`` node with the key ``name``, then that key of the map would be lost.
```

**Review by dbrumann — APPROVED** [review]
:+1:

File: components/config/definition.rst (hunk 1 above)
Comment (javiereguiluz) [reply components/config/definition.rst]: I made these changes while merging. Thanks.

**Comment by javiereguiluz** [comment]
@melkamar thanks and congrats on your first Symfony Docs contribution!

**Comment by dbrumann** [comment]
Oh, I didn't even notice that. Congrats @melkamar and thanks for contributing.

**Comment by melkamar** [comment]
Thanks guys, that was quick 😄

**Pull request merged**